Lexer: Warning when compiling `lexer.c`

I guess this is related to #632 

```
compiling ../../../../ext/herb/../../src/lexer.c
../../../../ext/herb/../../src/lexer.c:91:47: warning: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'uint32_t' (aka 'unsigned int') [-Wshorten-64-to-32]
   91 |       lexer->current_position = lexer->source.length;
      |                               ~ ~~~~~~~~~~~~~~^~~~~~
1 warning generated.
```

We might want to explicitly cast it or change the `length` to be a `uint32` as well. Or is there a reason why we kept that as a `size_t`?

/cc @timkaechele

--- a/src/include/util/hb_string.h
+++ b/src/include/util/hb_string.h
@@ -3,10 +3,11 @@
 
 #include <stdbool.h>
 #include <stddef.h>
+#include <stdint.h>
 
 typedef struct HB_STRING_STRUCT {
   char* data;
-  size_t length;
+  uint32_t length;
 } hb_string_T;
 
 hb_string_T hb_string_from_c_string(const char* null_terminated_c_string);

--- a/src/util/hb_string.c
+++ b/src/util/hb_string.c
@@ -8,7 +8,7 @@ hb_string_T hb_string_from_c_string(const char* null_terminated_c_string) {
   hb_string_T string;
 
   string.data = (char*) null_terminated_c_string;
-  string.length = strlen(null_terminated_c_string);
+  string.length = (uint32_t) strlen(null_terminated_c_string);
 
   return string;
 }

